pkg/pillar/cmd/monitor/messages.go

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2024 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package monitor
+
+import (
+	"github.com/lf-edge/eve/pkg/pillar/types"
+	uuid "github.com/satori/go.uuid"
+)
+
+type NodeStatus struct {
+	Server         string                   `json:"server,omitempty"`
+	NodeUuid       uuid.UUID                `json:"node_uuid,omitempty"`
+	Onboarded      bool                     `json:"onboarded"`
+	AppSummary     types.AppInstanceSummary `json:"app_summary,omitempty"`
+	ZedAgentStatus types.ZedAgentStatus     `json:"zedagent_status,omitempty"`
+}
+
+type AppInstancesStatus struct {
+	Apps []types.AppInstanceStatus `json:"apps"`
+}
+
+type NetworkStatus struct {
+}
+
+func (ctx *monitorContext) isOnboarded() (bool, uuid.UUID) {
+	sub := ctx.subscriptions["OnboardingStatus"]
+	if item, err := sub.Get("global"); err == nil {
+		onboardingStatus := item.(types.OnboardingStatus)
+		return true, onboardingStatus.DeviceUUID
+	}
+	return false, uuid.UUID{}
+}
+
+func (ctx *monitorContext) getAppSummary() types.AppInstanceSummary {
+	// send the network status to the server
+	sub := ctx.subscriptions["AppSummary"]
+	if item, err := sub.Get("global"); err == nil {
+		appSummary := item.(types.AppInstanceSummary)
+		return appSummary
+	}
+	return types.AppInstanceSummary{}
+}
+
+func (ctx *monitorContext) sendNodeStatus() {
+	// send the node status to the server
+	nodeStatus := NodeStatus{
+		Server: ctx.serverNameAndPort,
+	}
+	if onboarded, nodeUUID := ctx.isOnboarded(); onboarded {
+		nodeStatus.NodeUuid = nodeUUID
+		nodeStatus.Onboarded = true
+	}
+
+	nodeStatus.ZedAgentStatus = ctx.getZedAgentStatus()
+	nodeStatus.AppSummary = ctx.getAppSummary()
+
+	if ctx.lastNodeStatus != nil && *ctx.lastNodeStatus == nodeStatus {
```

Comment:
Is there a reason behind this check?

Reply:
@christoph-zededa I have a plan to reuse this function to send a node stats to abstract types used in EVE from types used in Go-Rust interface